packages/server-renderer/src/helpers/ssrRenderSlot.ts

```diff
@@ -83,7 +83,16 @@ export function ssrRenderSlotInner(
         }
       } else {
         for (let i = 0; i < slotBuffer.length; i++) {
-          push(slotBuffer[i])
+          const buffer = slotBuffer[i]
+          // #9933
+          // Although we handle Transition/TransitionGroup in the transform stage
+          // without rendering it as a fragment, the content passed into the slot
+          // may still be a fragment.
+          // Therefore, here we need to avoid rendering it as a fragment again.
+          if (transition && (buffer === '<!--[-->' || buffer === '<!--]-->')) {
```

Comment:
Is this safe? There might be nested fragments inside the slot buffer content. We should only unwrap the outer-most fragment.

Reply:
done!